File: TypeScript/1071. Greatest Common Divisor of Strings.ts
// noinspection JSUnusedGlobalSymbols

const gcdOfStrings = (str1: string, str2: string): string => {
  if (str1 + str2 !== str2 + str1) {
    return ''
  }

  const gcd = (a: number, b: number): number => (b === 0 ? a : gcd(b, a % b))
  const len = gcd(str1.length, str2.length)

  return str1.substring(0, len)
}

File: Go/1071. Greatest Common Divisor of Strings.go
package main

//goland:noinspection GoUnusedFunction
func gcdOfStrings(str1, str2 string) string {
	if str1+str2 != str2+str1 {
		return ""
	}

	return str1[:gcd(len(str1), len(str2))]
}

func gcd(former int, latter int) int {
	if latter == 0 {
		return former
	}

	return gcd(latter, former%latter)
}
